inginious/frontend/pages/utils.py

```diff
@@ -229,6 +232,35 @@ def preview_allowed(self, *args, **kwargs):
         return False
 
 
+class INGIniousAdministratorPage(INGIniousAuthPage):
+    """
+       Augmented version of INGIniousAuthPage that checks if user is administrator (superadmin).
+    """
+
+    def GET(self, *args, **kwargs):
+        """
+        Checks if user is superadmin and calls GET_AUTH or performs logout.
+        Otherwise, returns the login template.
+        """
+        username = self.user_manager.session_username()
+        if self.user_manager.session_logged_in():
+            if not self.user_manager.user_is_superadmin(username):
+                return self.template_helper.render("forbidden.html", message=_("You have not eligible right to see this part"))
```

Comment:
You can be eligible, but eligible right has no meaning. There should already be such forbidden access messages elsewhere that don't need to be repeated to ease localization.